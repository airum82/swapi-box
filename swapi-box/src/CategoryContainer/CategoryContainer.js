import React from 'react';
import { Card } from '../Card/Card';
import './CategoryContainer.css';
import PropTypes from 'prop-types';

export const CategoryContainer = ({ category }) => {
  let categoryCards;
  if(category.length && Object.keys(category[0]).includes('species')) {
    categoryCards = category.map(item => {
    return (
      <Card
        name={item.name}
        species={item.species}
        language={item.language}
        population={item.population}
      />
    )}
  )} else if(category.length && Object.keys(category[0]).includes('residents')) {
      categoryCards = category.map(item => {
        return (
        <Card
          name={item.name}
          population={item.population}
          terrain={item.terrain}
          climate={item.climate}
          residents={item.residents}
        />
      )}
    )} else if(category.length) {
        categoryCards = category.map(item => {
          return (
            <Card
              name={item.name}
              model={item.model}
              classType={item.classType}
              passengerNumber={item.passengerNumber}
            />
          )
        })
      }
  return (
    <section className="category-container">
    { categoryCards ? categoryCards : ''}
    </section>
  )
}

CategoryContainer.propTypes = {
  category: PropTypes.array
}