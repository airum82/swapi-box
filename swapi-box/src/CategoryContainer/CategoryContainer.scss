.category-container {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  background-color: gray

}

.category-container ul {
  list-style-type: none;
}

.category-container div {
  background-color: #7a2e2e;
}