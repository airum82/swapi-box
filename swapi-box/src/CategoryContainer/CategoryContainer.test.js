import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { CategoryContainer } from './CategoryContainer';

describe("CategoryContainer", () => {
  it('should match snapshot upon initial render', () => {
    const wrapper = shallow(<CategoryContainer
                      category={[]}
                    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when category has length and first item has a species key', () => {
    const mockCategory = [ { species: 'asdfsa',
                             name: 'leonard'},
                           { species: 'asdfasdf',
                             name: 'theo'}
                         ]
    const wrapper = shallow(<CategoryContainer
                      category={mockCategory}
                    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when category has length and first item has a residents key', () => {
    const mockCategory = [ { residents: [],
                             population: 4000},
                           { residents: [],
                             population: 49595}
                          ];
    const wrapper = shallow(<CategoryContainer
                             category={mockCategory}
                            />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when category container has length', () => {
    const mockCategory = [{ kind: 'does not matter'}, 
                          { kind: 'still does not'}
                         ];
    const wrapper = shallow(<CategoryContainer
                             category={mockCategory}
                            />);
    expect(wrapper).toMatchSnapshot();
  })
})