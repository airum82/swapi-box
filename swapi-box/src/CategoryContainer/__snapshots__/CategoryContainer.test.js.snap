// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryContainer should match snapshot upon initial render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    category={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "category-container",
    },
    "ref": null,
    "rendered": "",
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "category-container",
      },
      "ref": null,
      "rendered": "",
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CategoryContainer should match snapshot when category container has length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    category={
        Array [
            Object {
              "kind": "does not matter",
            },
            Object {
              "kind": "still does not",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          classType={undefined}
          model={undefined}
          name={undefined}
          passengerNumber={undefined}
/>,
        <Unknown
          classType={undefined}
          model={undefined}
          name={undefined}
          passengerNumber={undefined}
/>,
      ],
      "className": "category-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classType": undefined,
          "model": undefined,
          "name": undefined,
          "passengerNumber": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classType": undefined,
          "model": undefined,
          "name": undefined,
          "passengerNumber": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            classType={undefined}
            model={undefined}
            name={undefined}
            passengerNumber={undefined}
/>,
          <Unknown
            classType={undefined}
            model={undefined}
            name={undefined}
            passengerNumber={undefined}
/>,
        ],
        "className": "category-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "classType": undefined,
            "model": undefined,
            "name": undefined,
            "passengerNumber": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "classType": undefined,
            "model": undefined,
            "name": undefined,
            "passengerNumber": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CategoryContainer should match snapshot when category has length and first item has a residents key 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    category={
        Array [
            Object {
              "population": 4000,
              "residents": Array [],
            },
            Object {
              "population": 49595,
              "residents": Array [],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          climate={undefined}
          name={undefined}
          population={4000}
          residents={Array []}
          terrain={undefined}
/>,
        <Unknown
          climate={undefined}
          name={undefined}
          population={49595}
          residents={Array []}
          terrain={undefined}
/>,
      ],
      "className": "category-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "climate": undefined,
          "name": undefined,
          "population": 4000,
          "residents": Array [],
          "terrain": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "climate": undefined,
          "name": undefined,
          "population": 49595,
          "residents": Array [],
          "terrain": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            climate={undefined}
            name={undefined}
            population={4000}
            residents={Array []}
            terrain={undefined}
/>,
          <Unknown
            climate={undefined}
            name={undefined}
            population={49595}
            residents={Array []}
            terrain={undefined}
/>,
        ],
        "className": "category-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "climate": undefined,
            "name": undefined,
            "population": 4000,
            "residents": Array [],
            "terrain": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "climate": undefined,
            "name": undefined,
            "population": 49595,
            "residents": Array [],
            "terrain": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CategoryContainer should match snapshot when category has length and first item has a species key 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    category={
        Array [
            Object {
              "name": "leonard",
              "species": "asdfsa",
            },
            Object {
              "name": "theo",
              "species": "asdfasdf",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          language={undefined}
          name="leonard"
          population={undefined}
          species="asdfsa"
/>,
        <Unknown
          language={undefined}
          name="theo"
          population={undefined}
          species="asdfasdf"
/>,
      ],
      "className": "category-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "language": undefined,
          "name": "leonard",
          "population": undefined,
          "species": "asdfsa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "language": undefined,
          "name": "theo",
          "population": undefined,
          "species": "asdfasdf",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            language={undefined}
            name="leonard"
            population={undefined}
            species="asdfsa"
/>,
          <Unknown
            language={undefined}
            name="theo"
            population={undefined}
            species="asdfasdf"
/>,
        ],
        "className": "category-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "language": undefined,
            "name": "leonard",
            "population": undefined,
            "species": "asdfsa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "language": undefined,
            "name": "theo",
            "population": undefined,
            "species": "asdfasdf",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
